src/KubernetesClient/StreamDemuxer.cs

```diff
@@ -57,7 +57,7 @@ public StreamDemuxer(WebSocket webSocket, StreamType streamType = StreamType.Rem
         /// </summary>
         public void Start()
         {
-            this.runLoop = this.RunLoop(this.cts.Token);
+            this.runLoop = Task.Run(() => this.RunLoop(this.cts.Token));
```

Comment:
here the handle is to keep the task ref-ed by this obj and could be `waited` for in the future
in this case, `dispose` will wait for runloop to get things done and do some clean up.

wrapping with a task will lose the control to runloop task totally.
Task.Run will return immediately with an null

you may want `Task.Run(async() => await RunLoop())` if you want to keep sync context clean.
but I do not think it is necessary because no one wait on its parent obj

Reply:
> wrapping with a task will lose the control to runloop task totally. Task.Run will return immediately with an null

This change does not produce that behaviour.

As specified the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.run?view=netcore-2.0#System_Threading_Tasks_Task_Run_System_Func_System_Threading_Tasks_Task__), the `Task.Run(Func<Task>)` overload returns a proxy for the task returned by the provided `function`; therefore, the status of `runLoop` will reflect exactly that of the task returned by `RunLoop`.

We can observe this behaviour with [this](https://gist.github.com/andykernahan/a39fba0445e3936728f18447a6b7d841) `netcoreapp2.0` program which prints a value to stdout after simulating an asynchronous workload. As expected, the program prints the value and then exits.

> you may want `Task.Run(async() => await RunLoop())` if you want to keep sync context clean.
but I do not think it is necessary because no one wait on its parent obj

I'm afraid that I do not understand: making the suggested change does not affect the `SynchronizationContext` from any perspective.

The `function` is scheduled on `TaskScheduler.Default` which in turn executes it on the `ThreadPool` where there is no `SynchronizationContext`; therefore, there is no `SynchronizationContext` when `function` is invoked.

We can observe this behaviour with [this](https://gist.github.com/andykernahan/3f1a04a130098e3e358128b5220f676d) `netcoreapp2.0` program which prints `SynchronizationContext.Current`. As expected, the programs prints the expected `No SynchronizationContext`.

Reply:
Yes you are right 
at first I think your code is like below:
```
            var task = Task.Run(() =>
            {
                Task.Delay(TimeSpan.FromDays(1));
            });

            task.Wait(); // did not wait for the inner task
            Console.WriteLine("wait 1 d done"); 
```

`Task.Run(() => some task)` return and wait `some task` implicitly.

but if your code is like this 
```
            var task = Task.Run(async () =>
            {
                await Task.Delay(TimeSpan.FromDays(1));
            });
```
or
```
 Task.Run(async () =>await Task.Delay(TimeSpan.FromDays(1)));
```

you can keep bug away when editing the code in the future. 
That would be better for people to read the code.

Reply:
Let's use @tg123 's suggestion to make the code more readable and less compact.

Reply:
The maintainability aspect is interesting. However, converting the lambda to be statement bodied and omitting the `await` (as in your example) generates a CS4014 compiler warning due to the call not being awaited.

Would you like me to apply @tg123 's statement bodied suggestion or the expression bodied suggestion?

Reply:
hi
my suggestion is ` Task.Run(async () => await this.RunLoop(this.cts.Token));`

this one is show how bug happens
```
this.runLoop = Task.Run(() =>
{
     this.RunLoop(this.cts.Token); // bug forget await, same as your compiler warning
     // people wants to add something new
});
```

people might forget to await that `RunLoop`

while ` Task.Run(async () => await this.RunLoop(this.cts.Token));` will become

` Task.Run(async () => {
await this.RunLoop(this.cts.Token)); // never forget to await
// people wants to add something new
}`

my first look into the code is like `Task.Run( void () => ... )`. So I think it might be a bug.

Reply:
> this one is to show how bug happens, not suggestion

Yes, I understood that; I referred to that as your example, not suggestion.

I'm happy to make the suggested change of:

```csharp
Task.Run(async () => await this.RunLoop(this.cts.Token));
```

Reply:
will compiler give warning for this one?

Reply:
No. The compiler will not warn for any of the following:
```csharp
this.runLoop = Task.Run(() => this.RunLoop(this.cts.Token));
this.runLoop = Task.Run(async () => await this.RunLoop(this.cts.Token));
this.runLoop = Task.Run(async () =>
{
    await this.RunLoop(this.cts.Token);
});
```